cmd/commands/datasetIngestor.go

```diff
@@ -68,14 +68,57 @@ For Windows you need instead to specify -user username:password on the command l
 		userpass, _ := cmd.Flags().GetString("user")
 		token, _ := cmd.Flags().GetString("token")
 		copyFlag, _ := cmd.Flags().GetBool("copy")
-		nocopyFlag, _ := cmd.Flags().GetBool("nocopy") // TODO why is there even a "no copy" flag?
+		nocopyFlag, _ := cmd.Flags().GetBool("nocopy")
+		transferTypeFlag, _ := cmd.Flags().GetString("transfer-type")
 		tapecopies, _ := cmd.Flags().GetInt("tapecopies")
 		autoarchiveFlag, _ := cmd.Flags().GetBool("autoarchive")
 		linkfiles, _ := cmd.Flags().GetString("linkfiles")
 		allowExistingSourceFolder, _ := cmd.Flags().GetBool("allowexistingsource")
 		addAttachment, _ := cmd.Flags().GetString("addattachment")
 		addCaption, _ := cmd.Flags().GetString("addcaption")
 		showVersion, _ := cmd.Flags().GetBool("version")
+		globusCfgFlag, _ := cmd.Flags().GetString("globus-cfg")
+
+		// TODO: read in CFG!
+
+		// transfer type
+		transferType, err := convertToTransferType(transferTypeFlag)
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		var transferFiles func(params cliutils.TransferParams) (archivable bool, err error)
+
+		// globus specific vars (if needed)
+		var globusClient globus.GlobusClient
+		var srcCollection, srcPrefixPath, destCollection, destPrefixPath string
+
+		switch transferType {
+		case Ssh:
+			transferFiles = cliutils.SshTransfer
+		case Globus:
+			transferFiles = cliutils.GlobusTransfer
+			var globusConfigPath string
+			if cmd.Flags().Lookup("globus-cfg").Changed {
+				globusConfigPath = globusCfgFlag
+			} else {
+				execPath, err := os.Executable()
+				if err != nil {
+					log.Fatalln("can't find executable path:", err)
+				}
+				globusConfigPath = filepath.Join(filepath.Dir(execPath), "globus.yaml")
+			}
+
+			globusClient, srcCollection, srcPrefixPath, destCollection, destPrefixPath, err = cliutils.GlobusLogin(globusConfigPath)
+			if err != nil {
+				log.Fatalln("couldn't create globus client:", err)
+			}
+
+			if autoarchiveFlag {
+				log.Println("Cannot autoarchive when transferring via Globus due to the transfer happening asynchronously. Use the \"globusCheckTransfer\" command to archive them")
```

Comment:
Maybe should be Fatal to avoid user mistakes?

Reply:
Good idea I'll change it.